microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/faulttolerance/java/MicroProfileFaultToleranceASTValidator.java

```diff
@@ -274,21 +275,19 @@ private void validateRetryAnnotation(NormalAnnotation annotation)
 			Expression jitterUnitExpr = getAnnotationMemberValueExpression(
 					annotation, JITTER_DELAY_UNIT_RETRY_ANNOTATION_MEMBER);
 
-			Object delayConstant = delayExpr != null
-					? delayExpr.resolveConstantExpressionValue()
-					: null;
-			Object maxDurationConstant = maxDurationExpr != null
-					? maxDurationExpr.resolveConstantExpressionValue()
-					: null;
-			Object jitterConstant = jitterExpr != null
-					? jitterExpr.resolveConstantExpressionValue()
+			Object delayConstant = delayExpr != null ? delayExpr.resolveConstantExpressionValue() : null;
+			Object maxDurationConstant = maxDurationExpr != null ? maxDurationExpr.resolveConstantExpressionValue()
 					: null;
+			Object jitterConstant = jitterExpr != null ? jitterExpr.resolveConstantExpressionValue() : null;
 
-			int delayNum = delayConstant != null ? (int) delayConstant : -1;
-			int maxDurationNum = maxDurationConstant != null
-					? (int) maxDurationConstant
-					: -1;
-			int jitterNum = jitterConstant != null ? (int) jitterConstant : 0;
+			long delayNum = delayConstant instanceof Integer ? (long) (int) delayConstant

```

Comment:
Casting to an int, then a long after determining that the type is an integer is the equivalent to just casting to a long. Thus the `instanceof Long` check will always pass. To handle both cases separately, it might be better to check each condition separately then using a ternary assignment.